src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs

```diff
@@ -2471,5 +2471,19 @@ public MockConfigurationRoot(IList<IConfigurationProvider> providers) : base(pro
             IConfigurationSection IConfiguration.GetSection(string key) =>
                 this[key] is null ? null : new ConfigurationSection(this, key);
         }
+
+        [Fact]
+        public void CanBindToClassWithNewProperties()
+        {
+            var config = TestHelpers.GetConfigurationFromJsonString(@"{""A"":""Should not bind"", ""B"":""5"", ""E"":""OptionB""}");
```

Comment:
(minor) could use `"""` strings here for readabililty:

```suggestion
            var config = TestHelpers.GetConfigurationFromJsonString("""
                {
                  "A": "Should not bind",
                  "B": "5",
                  "E": "OptionB"
                 }
                 """);
```